builtin/logical/database/path_creds_create.go

```diff
@@ -51,7 +51,7 @@ func (b *databaseBackend) pathCredsCreateRead() framework.OperationFunc {
 		// If role name isn't in the database's allowed roles, send back a
 		// permission denied.
 		if !strutil.StrListContains(dbConfig.AllowedRoles, "*") && !strutil.StrListContainsGlob(dbConfig.AllowedRoles, name) {
-			return nil, logical.ErrPermissionDenied
+			return nil, fmt.Errorf("%s is not an allowed role", name)
```

Comment:
Minor preference for %q here, just to make it super clear

Reply:
Ah yes, that would read better, thanks!